components/gitpod-db/src/typeorm/workspace-db-impl.ts

```diff
@@ -1096,13 +1057,17 @@ export class TypeORMWorkspaceDBImpl extends TransactionalDBImpl<WorkspaceDB> imp
         return <WorkspaceAndInstance>res;
     }
 
-    async findInstancesByPhaseAndRegion(phase: string, region: string): Promise<WorkspaceInstance[]> {
+    async findInstancesByPhase(phases: string[]): Promise<WorkspaceInstance[]> {
```

Comment:
I wonder what is an implication of running this query each 10sec in each server on prod DB?

Reply:
It's not too much: "building" + "preparing" is always ~10 instances at any given point in time.

Pending can become more, around ~50-100, especially in the face of a workspace incident. This is the part I like the least about this approach; but again, can be fixed by a job.